src/solvers/refinement/string_constraint_generator_comparison.cpp

```diff
@@ -45,7 +45,9 @@ exprt string_constraint_generatort::add_axioms_for_equals(
 
   symbol_exprt qvar=fresh_univ_index("QA_equal", index_type);
   string_constraintt a2(
-    qvar, s1.length(), implies_exprt(eq, equal_exprt(s1[qvar], s2[qvar])));
+    qvar,
+    zero_if_negative(s1.length()),
+    implies_exprt(eq, equal_exprt(s1[qvar], s2[qvar])));
   constraints.push_back(a2);
```

Comment:
what I like to do is put the constraint definition (including the definition of qvar) in a lambda inside the push_back. This avoids reusing `qvar` in over formulas by mistake (which happened to me a couple of times)

Reply:
Encapsulated all three lemmas and constraint additions in this method. I will keep that in mind if I modify further constraints.